src/Dofs/DirichletBoundaryConditions2.jl

```diff
@@ -0,0 +1,342 @@
+    end
+    return vtkfile
+end
+
+function apply!(v::Vector, dbcs::DirichletBoundaryConditions)
+    @assert length(v) == ndofs(dbcs.dh)
+    v[dbcs.prescribed_dofs] = dbcs.values # .= ??
+    return v
+end
+
+function apply_zero!(v::Vector, dbcs::DirichletBoundaryConditions)
+    @assert length(v) == ndofs(dbcs.dh)
+    v[dbcs.prescribed_dofs] = 0 # .= ?
+    return v
+end
+
+function apply!(K::Union{SparseMatrixCSC, Symmetric}, dbcs::DirichletBoundaryConditions)
+    apply!(K, eltype(K)[], dbcs, true)
+end
+
+function apply_zero!(K::Union{SparseMatrixCSC, Symmetric}, f::AbstractVector, dbcs::DirichletBoundaryConditions)
+    apply!(K, f, dbcs, true)
+end
+
+function apply!(KK::Union{SparseMatrixCSC, Symmetric}, f::AbstractVector, dbcs::DirichletBoundaryConditions, applyzero::Bool=false)
+    K = isa(KK, Symmetric) ? KK.data : KK
+    @assert length(f) == 0 || length(f) == size(K, 1)
+    @boundscheck checkbounds(K, dbcs.prescribed_dofs, dbcs.prescribed_dofs)
+    @boundscheck length(f) == 0 || checkbounds(f, dbcs.prescribed_dofs)
+
+    m = meandiag(K) # Use the mean of the diagonal here to not ruin things for iterative solver
+    @inbounds for i in 1:length(dbcs.values)
+        d = dbcs.prescribed_dofs[i]
+        v = dbcs.values[i]
+
+        if !applyzero && v != 0
+            for j in nzrange(K, d)
+                f[K.rowval[j]] -= v * K.nzval[j]
+            end
+        end
+    end
+    K[:, dbcs.prescribed_dofs] = 0
+    K[dbcs.prescribed_dofs, :] = 0
+    @inbounds for i in 1:length(dbcs.values)
+        d = dbcs.prescribed_dofs[i]
+        v = dbcs.values[i]
+        K[d, d] = m
+        # We will only enter here with an empty f vector if we have assured that v == 0 for all dofs
+        if length(f) != 0
+            vz = applyzero ? zero(eltype(f)) : v
+            f[d] = vz * m
+        end
+    end
+end
+
+function meandiag(K::AbstractMatrix)
+    z = zero(eltype(K))
+    for i in 1:size(K, 1)
+        z += abs(K[i, i])
+    end
+    return z / size(K, 1)
+end
+
+create_field_facevalue(func_interpol::Interpolation, geom_interpol::Interpolation) =
+    create_field_facevalue(Float64, func_interpol, geom_interpol)
+
+function create_field_facevalue(::Type{T}, func_interpol::Interpolation{dim,refshape}, geom_interpol::Interpolation{dim,refshape}) where {T,dim,refshape}
+    # set up quadrature rules for each face with dof-positions
+    # (determined by func_interpol) as the quadrature points
+    interpolation_coords = reference_coordinates(func_interpol)
+
+    qrs = QuadratureRule{dim,refshape,T}[]
+    for face in faces(func_interpol)
+        dofcoords = Vec{dim,T}[]
+        for facedof in face
+            push!(dofcoords, interpolation_coords[facedof])
+        end
+        qrf = QuadratureRule{dim,refshape,T}(fill(T(NaN), length(dofcoords)), dofcoords) # weights will not be used
+        push!(qrs, qrf)
+    end
+
+    # TODO: The rest is just to fill up the FaceScalarValues, can be removed
```

Comment:
We can have a separate struct for this maybe, all that we need is `M`

Reply:
Yeah, seems reasonable when so much is just garbage to actually create the struct.

Reply:
`BoundaryConditionValues` incoming.